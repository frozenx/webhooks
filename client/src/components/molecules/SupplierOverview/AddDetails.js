import React from "react";
import { IconButtonLinkWithText } from '@beans/link';
import { AddDetailsWrapper } from "./styled";

const AddDetails = props => {
    return (
        <AddDetailsWrapper>
            <IconButtonLinkWithText
                icon={{ graphic: 'add', size: 'xs', position: { global: 'left' } }}
                buttonVariant="secondary"
                disabled={!props.canAddNewSite}
                emphasized={true}
                onClick={() => { props.addNewSite() }}
            >
                {props.title}
            </IconButtonLinkWithText>
        </AddDetailsWrapper>
    );
}

export default AddDetails;
