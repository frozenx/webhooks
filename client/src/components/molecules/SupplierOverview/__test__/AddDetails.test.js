import React from "react";
import { shallow } from "enzyme";
import AddDetails from "../AddDetails";

describe("add site & payment card component test cases", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddDetails />);
  });

  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
