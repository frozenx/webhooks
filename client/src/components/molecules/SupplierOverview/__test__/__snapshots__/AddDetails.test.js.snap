// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add site & payment card component test cases render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <_default
          className="site-title"
          dark={true}
          emphasized={true}
        >
           Site details
        </_default>,
        <ForwardRef
          buttonVariant="secondary"
          disabled={true}
          emphasized={true}
          icon={
            Object {
              "graphic": "add",
              "position": Object {
                "global": "left",
              },
              "size": "xs",
            }
          }
        >
          Add Site Details
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": " Site details",
          "className": "site-title",
          "dark": true,
          "emphasized": true,
        },
        "ref": null,
        "rendered": " Site details",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonVariant": "secondary",
          "children": "Add Site Details",
          "disabled": true,
          "emphasized": true,
          "icon": Object {
            "graphic": "add",
            "position": Object {
              "global": "left",
            },
            "size": "xs",
          },
        },
        "ref": null,
        "rendered": "Add Site Details",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "componentName": "IconButtonLinkWithText",
          "displayName": "WithProps(IconButtonLinkWithText)",
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": true,
        "rules": Array [
          "
    > a{
        width:100%;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <_default
            className="site-title"
            dark={true}
            emphasized={true}
          >
             Site details
          </_default>,
          <ForwardRef
            buttonVariant="secondary"
            disabled={true}
            emphasized={true}
            icon={
              Object {
                "graphic": "add",
                "position": Object {
                  "global": "left",
                },
                "size": "xs",
              }
            }
          >
            Add Site Details
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": " Site details",
            "className": "site-title",
            "dark": true,
            "emphasized": true,
          },
          "ref": null,
          "rendered": " Site details",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonVariant": "secondary",
            "children": "Add Site Details",
            "disabled": true,
            "emphasized": true,
            "icon": Object {
              "graphic": "add",
              "position": Object {
                "global": "left",
              },
              "size": "xs",
            },
          },
          "ref": null,
          "rendered": "Add Site Details",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "componentName": "IconButtonLinkWithText",
            "displayName": "WithProps(IconButtonLinkWithText)",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "rules": Array [
            "
    > a{
        width:100%;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
